spec/forms/hyrax/forms/permission_template_form_spec.rb

```diff
@@ -195,6 +220,95 @@
     end
   end
 
+  describe "#validate_visibility_combinations" do
+    let(:admin_set) { create(:admin_set) }
+    let(:permission_template) { create(:permission_template, admin_set_id: admin_set.id) }
+    let(:form) { described_class.new(permission_template) }
+
+    context "validate all release option attribute combinations" do
+      def self.expect_attributes_to_be_valid(attributes = {})
+        it "will be valid? for #{attributes.inspect}" do
+          attributes = ActionController::Parameters.new(**attributes).permit!
+          expect(form.update(attributes)).to eq(content_tab: "visibility", updated: true)
+        end
+      end
+
+      def self.expect_attributes_not_to_be_valid(attributes = {})
+        it "will not be valid? for #{attributes.inspect}" do
+          attributes = ActionController::Parameters.new(**attributes).permit!
+          subject = form.update(attributes)
+          expect(subject).to include(content_tab: "visibility", updated: false)
+          expect(I18n.t(subject[:error_code], scope: 'hyrax.admin.admin_sets.form.permission_update_errors')).not_to include('translation missing')
+        end
+      end
+
+      # no delay
+      expect_attributes_to_be_valid(
+        visibility: '',
+        release_date: Time.zone.today,
+        release_period: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_NO_DELAY,
+        release_varies: nil,
+        release_embargo: nil
+      )
+      # varies, with date selected
+      expect_attributes_to_be_valid(
+        visibility: '',
+        release_date: Time.zone.today + 2.months,
+        release_period: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_BEFORE_DATE,
+        release_varies: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_BEFORE_DATE,
+        release_embargo: nil
+      )
+      # varies, with embargo selected
+      expect_attributes_to_be_valid(
+        visibility: '',
+        release_date: Time.zone.today + 2.years,
+        release_period: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_2_YEARS,
+        release_varies: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_EMBARGO,
+        release_embargo: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_2_YEARS
+      )
+      # fixed, with date selected
+      expect_attributes_to_be_valid(
+        visibility: '',
+        release_date: Time.zone.today + 2.months,
+        release_period: Hyrax::PermissionTemplate::RELEASE_TEXT_VALUE_FIXED,
+        release_varies: nil,
+        release_embargo: nil
+      )
+      # varies, but no subsequent options
+      expect_attributes_not_to_be_valid(
```

Comment:
This case is the same as `varies, with embargo option but no period` at line 294, i.e. all empty string/nil values!